runtime-infra/log4ses.yaml

```diff
@@ -2,44 +2,77 @@ AWSTemplateFormatVersion: '2010-09-09'
+          Protocol: lambda
+          Endpoint: !GetAtt LambdaFunction.Arn
+          TopicArn: !Ref SesSnsTopic
+  LambdaInvokePermission:
+      Type: AWS::Lambda::Permission
+      Properties:
+          Action: lambda:InvokeFunction
+          Principal: sns.amazonaws.com
+          SourceArn:  !Ref SesSnsTopic
+          FunctionName: !Ref LambdaFunction
+  LambdaFunctionRole:
+    Type: AWS::IAM::Role
     Properties:
       AssumeRolePolicyDocument:
-        Version: "2012-10-17"
+        Version: '2012-10-17'
         Statement:
-          - Effect: "Allow"
-            Principal:
-              Service: "pipes.amazonaws.com"
-            Action: "sts:AssumeRole"
+        - Effect: Allow
+          Principal:
+            Service:
+              - lambda.amazonaws.com
+          Action:
+            - sts:AssumeRole
+      Path: "/"
       Policies:
-        - PolicyName: "BaseInlinePolicy"
-          PolicyDocument:
-            Version: "2012-10-17"
-            Statement:
-              - Sid: "AccessToCloudWatch"
-                Effect: "Allow"
-                Action:
-                  - "logs:CreateLogGroup"
-                  - "logs:CreateLogStream"
-                  - "logs:PutLogEvents"
-                Resource: "*"
-              - Sid: "UseSqsAsSource"
-                Effect: "Allow"
-                Action:
-                  - "sqs:ReceiveMessage"
-                  - "sqs:DeleteMessage"
-                  - "sqs:GetQueueAttributes"
-                Resource: !GetAtt "SesSqsQueue.Arn"
-  SesLog:
+      - PolicyName: LambdaFunctionPolicy
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+          - Effect: Allow
+            Action:
+              - logs:CreateLogGroup
+              - logs:CreateLogStream
+              - logs:PutLogEvents
+            Resource: '*'
+  LambdaFunctionLogGroup:
     Type: AWS::Logs::LogGroup
+    DependsOn: LambdaFunction
+    Properties:
+      RetentionInDays: 14
+      LogGroupName: !Join
+        - ""
+        - -  "/aws/lambda/"
+          - !Ref LambdaFunction
+  LambdaFunction:
+    Type: AWS::Lambda::Function
     Properties:
-      RetentionInDays: "14"
-      LogGroupName: "/aws/events/ses"
+      Runtime: python3.10
+      Timeout: 5
+      Handler: index.lambda_handler
+      Role: !GetAtt LambdaFunctionRole.Arn
+      Code:
+        ZipFile:
+          !Sub
+            - |-
+              import re
+
+              def replace_email(string):
+                  pattern = r'\b([A-Za-z0-9._%+-]+)@([A-Za-z0-9.-]+\.[A-Za-z]{2,})\b'
+                  return re.sub(pattern, r'***@\2', string)
+
+              def lambda_handler(event, context):
+                  single_event = event['Records'][0]['Sns']['Message']
+                  single_event = replace_email(single_event)
 
+                  print(single_event)
+                  return single_event
+            -
+              lambda_function_role_arn: !Ref LambdaFunctionRole
```

Comment:
@mbrachi non capisco questa riga, a cosa serve?

Reply:
tolta la stringa, in quanto i ruoli sono specificati poco sopra